### PR TITLE
Heretic Target Selection Dialog Tells you Which Possible Targets are on the Same Team as You

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
## About The Pull Request
When selecting your sacrifice target, the selection dialog will let you know if you're on the same antag team(s) as them by appending their name/job with "(ally)".

edit: Target selection also deprioritizes teammates, meaning that teammates can only be selected as sacrifice targets if there aren't enough possible sacrifice targets otherwise. If this is considered out of scope, the commit responsible will be removed and, if this one is merged, made into its own PR.

## Why It's Good For The Game
I was originally planning to make it so you cannot select people you share an antag team with as sacrifice targets because of the server rules requiring consent in order to sacrifice fellow cultists/revs, but after discussing the subject with @MrMelbert, I decided a fair compromise would be letting heretics know which of the targets they get to pick from are allies.

## Changelog
:cl:
qol: When selecting a sacrifice target, heretics that are also on an antag team(s) (E.G. cult, revs) will be told which of their potential targets are also on the same team(s) as them.
/:cl:

### CODE REVIEW

**Comment by Sparkezel** [comment]
The rules don't really for I'd you from sacrificing them, just ask for their consent really

**Comment by InsaneRed** [comment]
you see mr eldritch god i worship i simply cannot sacrifice cute catgirl number#5252052502052 because we dislike the heads of staff

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:287]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:294]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:300]: Why don't we just change the target selection algorithm to prioritize people not on your team first, only giving you people on your team when all other options are exhausted?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (ATH1909) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:300]: > Why don't we just change the target selection algorithm to prioritize people not on your team first, only giving you people on your team when all other options are exhausted?

if you do this, please keep the (ally) thing, for those rare scenarios where there aren't enough non-allied potential targets to fill out a whole list.

**Comment by Y0SH1M4S73R** [comment]
I decided that deprioritizing teammates from getting selected as heretic targets is out of scope. I moved the commit to its own branch that will get PRed if this gets merged.

**Comment by Mothblocks** [comment]
Your call, I didn't think it was personally. 🤷

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:286]: Last comment--make these `as anything`, as in:

```suggestion
			for(var/datum/antagonist/antag as anything in user.mind.antag_datums)
```

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (Mothblocks) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:294]:
```suggestion
				for(var/datum/team/team as anything in teams)
```

Didn't pick this up the first time, my mistake.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
:+1: LGTM, undraft if ready.

**Pull request merged**
